cumulusci/tasks/salesforce/custom_settings.py

```diff
@@ -0,0 +1,124 @@
+            self.options.get("settings_path")
+        )
+        if self.options["settings_path"] is None or not os.path.isfile(
+            self.options["settings_path"]
+        ):
+            raise TaskOptionsError(
+                f"File {self.options['settings_path']} does not exist"
+            )
+
+    def _run_task(self):
+        with open(self.options["settings_path"], "r") as f:
+            self.settings = yaml.safe_load(f)
+
+        self._load_settings()
+
+    def _load_settings(self):
+        # For each top-level heading in our YAML doc, create one or more
+        # custom settings.
+
+        for custom_setting, settings_data in self.settings.items():
+            proxy_obj = getattr(self.sf, custom_setting)
+            # If this level is a dict, we're working with a List Custom Setting
+            # If it's a list, we have a Hierarchy Custom Setting.
+            if isinstance(settings_data, dict):
+                for setting_instance, instance_data in settings_data.items():
+                    proxy_obj.upsert("Name/{}".format(setting_instance), instance_data)
+            elif isinstance(settings_data, list):
+                for setting_instance in settings_data:
+                    query = None
+
+                    if "location" in setting_instance:
+                        if "profile" in setting_instance["location"]:
+                            # Query for a matching Profile to assign the Setup Owner Id.
+                            profile_name = setting_instance["location"]["profile"]
+                            query = (
+                                f"SELECT Id FROM Profile WHERE Name = '{profile_name}'"
+                            )
+                        elif "user" in setting_instance["location"]:
+                            if "name" in setting_instance["location"]["user"]:
+                                # Query for a matching User to assign the Setup Owner Id.
+                                user_name = setting_instance["location"]["user"]["name"]
+                                query = f"SELECT Id FROM User WHERE Username = '{user_name}'"
+                            elif "email" in setting_instance["location"]["user"]:
+                                # Query for a matching User to assign the Setup Owner Id.
+                                email_address = setting_instance["location"]["user"][
+                                    "email"
+                                ]
+                                query = f"SELECT Id FROM User WHERE Email = '{email_address}'"
+                        elif "org" == setting_instance["location"]:
+                            # Assign the Setup Owner Id to the organization.
+                            query = "SELECT Id FROM Organization"
+
+                    if query is None:
+                        raise CumulusCIException(
+                            f"No valid Setup Owner assignment found for Custom Setting {custom_setting}"
```

Comment:
Can this be rewritten to give the user a clue about how they could resolve the problem?